src/descriptor/mod.rs

```diff
@@ -852,10 +847,7 @@ mod test {
         let result = into_wallet_descriptor_checked(descriptor, &secp, Network::Testnet);
 
         assert!(result.is_err());
-        assert!(matches!(
-            result.unwrap_err(),
-            DescriptorError::HardenedDerivationXpub
-        ));
+        assert_matches!(result.unwrap_err(), DescriptorError::HardenedDerivationXpub);
```

Comment:
Can be written as
```
        assert_matches!(result, Err(DescriptorError::HardenedDerivationXpub));
```

and also you can drop the `assert!` in the line above. I think it was there because `assert!(matches!(` is not really informative when there's a failure - but the error reporting of `assert_matches` is fine, so we don't need those tricks anymore :)